inc/vcf/fixer.hpp

```diff
@@ -108,9 +109,47 @@ namespace ebi
             util::writeline(output, *line);
             ignored_errors++;
         }
+        /**
+         * fix: remove the info field that caused the error
+         */
         virtual void visit(InfoBodyError &error) override
         {
-            ignored_errors++;
+            // TODO better log system, if any
+            std::cerr << "DEBUG: line " << error.get_line() << ": fixing invalid INFO field " << error.get_field()
+                      << std::endl;
+            const int info_column_index = 7;
+            const std::string empty_info_column = ".";
+
+            size_t num_removed_subfields = 0;
+            std::string string_line = {line->begin(), line->end()};
+
+            fix_column(info_column_index, string_line, "\t", [&](std::string &info_column) {
+                std::vector<std::string> info_subfields, keys_values;
+                util::string_split(info_column, ";", info_subfields);
+
+                for (size_t j = 0; j < info_subfields.size(); ++j) {
+                    util::string_split(info_subfields[j], "=", keys_values);
+                    if (keys_values[0] != error.get_field()) {
+                        if ((num_removed_subfields - j) != 0) {
```

Comment:
These 2 fields being of type `size_t`, subtracting them could cause overflow issues. Comparing one against the other should be fine.